test/unified_memory_allocation/uma_pools/disjoint_pool_config.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include "helpers.hpp"
+#include "usm_pool_config.hpp"
+
+using PoolDescriptor = usm_pool::PoolDescriptor;
+
+const PoolDescriptor DESCRIPTORS[4] = {PoolDescriptor(UR_USM_TYPE_HOST),
+                                       PoolDescriptor(UR_USM_TYPE_DEVICE),
+                                       PoolDescriptor(UR_USM_TYPE_SHARED),
+                                       PoolDescriptor(UR_USM_TYPE_SHARED, true)};
+
+TEST(disjointPoolConfigTests, disjointPoolConfigTest) {
+    // test if the returned unordered map contains only the valid configs
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>();
+
+    // check if all the valid configs are present
+    for (const auto &descriptor : DESCRIPTORS) {
+        ASSERT_EQ(configsMap.count(descriptor), 1);
+    }
+
+    // check if there are no more configs in the unordered map
+    ASSERT_EQ(configsMap.size(), 4);
+}
+
+TEST(disjointPoolConfigTests, disjointPoolConfigStringEnabledBuffersTest) {
+    // test for valid string with enabled buffers-- (values to configs should be
+    // parsed from string)
+    std::string config = "1;32M;host:1M,4,64k;device:1m,4,64K;shared:0,3,1M;"
+                         "read_only_shared:0,0,3M";
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>(config);
+
+    // test for host
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).MaxPoolableSize, 1 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).SlabMinSize, 64 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).limits->MaxSize, 32 * 1024 * 1024);
+
+    // test for device
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).MaxPoolableSize, 1 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).SlabMinSize, 64 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).limits->MaxSize, 32 * 1024 * 1024);
+
+    // test for shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).MaxPoolableSize, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).Capacity, 3);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).SlabMinSize, 1 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).limits->MaxSize, 32 * 1024 * 1024);
+
+    // test for read-only shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).MaxPoolableSize, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).Capacity, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).SlabMinSize, 3 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).limits->MaxSize, 32 * 1024 * 1024);
+}
+
+TEST(disjointPoolConfigTests, disjointPoolConfigStringImpartialEnabledBuffersTest) {
+    // test for valid impartial string with enabled buffers-- (values to configs should be
```

Comment:
Add tests for partially valid and partially invalid strings.
If the string passed is case-sensitive, add tests for typos such as `k->K`, `M->m`.
Add a test where a wrong type of memory is provided, like: `foo:1,4,64`.

Reply:
Done. I've added those tests. The passed string is not case-sensitive so I used both lower and upper case letters in test cases.